3.8/aql/graphs-k-paths.md

```diff
@@ -0,0 +1,239 @@
+  values will lead to a warning and an empty result. If one of the specified
+  documents does not exist, the result is empty as well and there is no warning.
+- `GRAPH` **graphName** (string): the name identifying the named graph. Its vertex and
+  edge collections will be looked up.
+- `OPTIONS` **options** (object, *optional*): used to modify the execution of the
+  search. Right now there are no options that trigger an effect. However this may change in the future.
+
+### Working with collection sets
+
+```
+FOR path
+  IN OUTBOUND|INBOUND|ANY K_PATHS
+  startVertex TO targetVertex
+  edgeCollection1, ..., edgeCollectionN
+  [OPTIONS options]
+```
+
+Instead of `GRAPH graphName` you can specify a list of edge collections.
+The involved vertex collections are determined by the edges of the given
+edge collections. 
+
+### Traversing in mixed directions
+
+For k paths with a list of edge collections you can optionally specify the
+direction for some of the edge collections. Say for example you have three edge
+collections *edges1*, *edges2* and *edges3*, where in *edges2* the direction
+has no relevance, but in *edges1* and *edges3* the direction should be taken into
+account. In this case you can use `OUTBOUND` as general search direction and `ANY`
+specifically for *edges2* as follows:
+
+```
+FOR vertex IN OUTBOUND K_PATHS
+  startVertex TO targetVertex
+  edges1, ANY edges2, edges3
+```
+
+All collections in the list that do not specify their own direction will use the
+direction defined after `IN` (here: `OUTBOUND`). This allows to use a different
+direction for each collection in your path search.
+
+Examples
+--------
+
+We load an example graph to get a named graph that reflects some possible
+train connections in Europe and North America.
+
+![Train Connection Map](../images/train_map.png)
+
+{% arangoshexample examplevar="examplevar" script="script" result="result" %}
+    @startDocuBlockInline GRAPHKP_01_create_graph
+    @EXAMPLE_ARANGOSH_OUTPUT{GRAPHKP_01_create_graph}
+    ~addIgnoreCollection("places");
+    ~addIgnoreCollection("connections");
+    var examples = require("@arangodb/graph-examples/example-graph.js");
+    var graph = examples.loadGraph("kShortestPathsGraph");
+    db.places.toArray();
+    db.connections.toArray();
+    @END_EXAMPLE_ARANGOSH_OUTPUT
+    @endDocuBlock GRAPHKP_01_create_graph
+{% endarangoshexample %}
+{% include arangoshexample.html id=examplevar script=script result=result %}
+
+Suppose we want to query all routes from **Aberdeen** to **London**.
+
+{% aqlexample examplevar="examplevar" type="type" query="query" bind="bind" result="result" %}
+    @startDocuBlockInline GRAPHKP_01_Aberdeen_to_London
+    @EXAMPLE_AQL{GRAPHKP_01_Aberdeen_to_London}
+    @DATASET{kShortestPathsGraph}
+    FOR p IN OUTBOUND K_PATHS 'places/Aberdeen' TO 'places/London'
+    GRAPH 'kShortestPathsGraph'
+        RETURN { places: p.vertices[*].label, travelTimes: p.edges[*].travelTime }
```

Comment:
That query produces an empty result, because the limit is implicitly set to `1..1`. `1..10` yields for results and `1..5` yields one.

Reply:
You are right, I totally forgot to add the min..max depths...